docs/src/writing-shader-crates.md

```diff
@@ -0,0 +1,72 @@
+# Writing Shader Crates
+
+This is section is going to walk you through writing a shader in Rust and
+setting up your shader crate.
+
+Be aware that this project is in a very early phase, please [file an
+issue](https://github.com/EmbarkStudios/rust-gpu/issues) if there's something
+not working or unclear.
+
+## Setup
+There are two main ways to setup your shader project.
+
+1. Using the `spirv-builder` crate.
+   The `spirv-builder` is a crate designed to automate the process of building
+   and linking the `rust-gpu` to be able to compile SPIR-V shaders into your
+   main Rust crate.
+2. Using `.cargo/config`.
+   Alternatively if you're willing to do the setup yourself you can manually set
+   flags in your cargo configuration to enable you to run `cargo build` in your
+   shader crate.
+
+
+### Using `spirv-builder`
+If you're writing a bigger application and you want to integrate SPIR-V shader
+crates to display, it's recommended to use `spirv-builder` in a build script.
+
+#### `build.rs`
+```rust,no_run
+SpirvBuilder::new(path_to_shader)
+        .spirv_version(1, 0)
+        .print_metadata()
+        .build()?;
+```
+
+#### `main.rs`
+```rust,no_run
+const SHADER: &[u8] = include_bytes!(env!("<shader_name>.spv"));
+```
+
+### Using `.cargo/config`
+If you just want to compile a build a shader crate, and don't need to
+automatically compile the SPIR-V binary at build time, you can use
+`.cargo/config` to set the necessary flags. Before you can do that however you
+need to do a couple of steps first to build the compiler backend.
+
+1. Clone the `rust-gpu` repository
+2. `cargo build --release`
+
+Now you should have a `librustc_codegen_spirv` dynamic library available in
+`target/release`. You'll need to keep this somewhere stable that you can
+reference from your shader project.
+
+Now we need to add our `.cargo/config` file. This tells cargo to build for the
+`spirv-unknown-unknown` target, and provides a path to the codegen backend for
+that target.
+
+```toml
+[build]
+target = "spirv-unknown-unknown"
+rustflags = ["-Zcodegen-backend=<path_to_librustc_codegen_spirv>"]
+```
+
+Now we can build our crate with cargo as normal. We have to provide
+`-Zbuild-std` as the `spirv-unknown-unknown` sysroot is not currently
+available in the default installation.
+
+```bash
+cargo build -Zbuild-std=core
```

Comment:
This can be specified in `.cargo/config` as well, we should put it there. I think it's something like

```toml
[unstable]
build-std = ["core"]
```

but my memory is fuzzy so we should double-check.

Reply:
Yep, I was looking for that, but couldn't find until I search for unstable specifically.